hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java

```diff
@@ -69,8 +69,9 @@ public void validate(YamlMapping rootNode) {
             // this could be expressed in the schema as well, but that would make all the schema validation errors much harder
             // to read, so it is better to implement it here as a semantic check
             long definedRootNodeCount = PERMITTED_ROOT_NODES.stream()
-                    .filter(rootNodeName -> rootNode.child(rootNodeName) != null)
+                    .filter(rootNodeName -> rootNode != null && rootNode.child(rootNodeName) != null)
```

Comment:
The fix looks fine to me. It would be good to have tests for the failing cases in the `Yaml*ConfigBuilderTest` classes. @fkardashov can you add them?

Reply:
ok. I can

Reply:
added some negative cases.
sorry, but now that's all